homeassistant/components/nextdns/switch.py

```diff
@@ -0,0 +1,246 @@
+    ),
+    NextDnsSwitchEntityDescription[Settings](
+        key="google_safe_browsing",
+        name="Google safe browsing",
+        entity_category=EntityCategory.CONFIG,
+        icon="mdi:google",
+        state=lambda data: data.google_safe_browsing,
+    ),
+    NextDnsSwitchEntityDescription[Settings](
+        key="idn_homograph_attacks_protection",
+        name="IDN homograph attacks protection",
+        entity_category=EntityCategory.CONFIG,
+        state=lambda data: data.idn_homograph_attacks_protection,
+    ),
+    NextDnsSwitchEntityDescription[Settings](
+        key="threat_intelligence_feeds",
+        name="Threat intelligence feeds",
+        entity_category=EntityCategory.CONFIG,
+        state=lambda data: data.threat_intelligence_feeds,
+    ),
+    NextDnsSwitchEntityDescription[Settings](
+        key="typosquatting_protection",
+        name="Typosquatting protection",
+        entity_category=EntityCategory.CONFIG,
+        icon="mdi:keyboard-outline",
+        state=lambda data: data.typosquatting_protection,
+    ),
+    NextDnsSwitchEntityDescription[Settings](
+        key="block_bypass_methods",
+        name="Block bypass methods",
+        entity_category=EntityCategory.CONFIG,
+        state=lambda data: data.block_bypass_methods,
+    ),
+    NextDnsSwitchEntityDescription[Settings](
+        key="safesearch",
+        name="Force SafeSearch",
+        entity_category=EntityCategory.CONFIG,
+        icon="mdi:search-web",
+        state=lambda data: data.safesearch,
+    ),
+    NextDnsSwitchEntityDescription[Settings](
+        key="youtube_restricted_mode",
+        name="Force YouTube restricted mode",
+        entity_category=EntityCategory.CONFIG,
+        icon="mdi:youtube",
+        state=lambda data: data.youtube_restricted_mode,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add NextDNS entities from a config_entry."""
+    coordinator: NextDnsSettingsUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        ATTR_SETTINGS
+    ]
+
+    switches: list[NextDnsSwitch] = []
+    for description in SWITCHES:
+        switches.append(NextDnsSwitch(coordinator, description))
+
+    async_add_entities(switches)
+
+
+class NextDnsSwitch(CoordinatorEntity[NextDnsSettingsUpdateCoordinator], SwitchEntity):
+    """Define an NextDNS switch."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: NextDnsSettingsUpdateCoordinator,
+        description: NextDnsSwitchEntityDescription,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+        self._attr_device_info = coordinator.device_info
+        self._attr_unique_id = f"{coordinator.profile_id}_{description.key}"
+        self._attr_is_on = description.state(coordinator.data)
+        self.entity_description: NextDnsSwitchEntityDescription = description
```

Comment:
The type it not needed here. If you want to set the type you should instead define the type on the class attribute.
```suggestion
        self.entity_description = description
```

```python
class NextDnsSwitch(CoordinatorEntity[NextDnsSettingsUpdateCoordinator], SwitchEntity):
    """Define an NextDNS switch."""

    _attr_has_entity_name = True
    entity_description: NextDnsSwitchEntityDescription
```